uint64 id
float64 desired_speed
